src/Components/HomePage.js

```diff
@@ -0,0 +1,105 @@
+import { useDispatch, useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { NavLink } from 'react-router-dom';
+import { MdSettings } from 'react-icons/md';
+import { FaMicrophone } from 'react-icons/fa';
+import { BsArrowRightCircle } from 'react-icons/bs';
+import { getAllCoins, getCoinMarkets } from '../Redux/CoinAPI';
+import cryptoPix from './images/crypto-pix.jpg';
+
+const HomePage = () => {
+  const getList = useSelector((state) => state.cryptoCoins);
+
+  const coins = getList.coin;
+  const { currency } = getList;
+
+  const dispatch = useDispatch();
+
+  const cur = 'USD';
+
+  useEffect(() => {
+    if (!coins) {
+      dispatch(getAllCoins(cur));
+    }
+  }, []);
+
+  const onSelect = (e) => {
+    dispatch(getAllCoins(e.target.value));
+  };
+
+  const handleClick = (e) => {
+    dispatch(getCoinMarkets(e.target.id));
+  };
+
+  return (
+    <div className="pg-container">
+      <nav>
+        <p className="heading">Top Cryptos</p>
+        <div className="nav-icon">
+          <FaMicrophone style={{ color: 'white', marginRight: '2rem' }} />
+          <MdSettings style={{ color: 'white' }} />
+        </div>
+      </nav>
+      <div className="pg-head">
+        <div className="crypto-pix">
+          <img className="image" src={cryptoPix} alt="currency" />
+        </div>
+        <div className="head-desr">
+          <h1 className="crypto-x">CRYPTO EXCHANGE</h1>
+        </div>
+
+      </div>
+
```

Comment:
- [ ] It would be better if you add a search bar that can filter currencies by their names as in the requirements so Please I encourage you to do that